js/deribit.js

```diff
@@ -1503,17 +1511,24 @@ module.exports = class deribit extends Exchange {
         //         "trades": [], // injected by createOrder
         //     }
         //
+        const marketId = this.safeString (order, 'instrument_name');
+        market = this.safeMarket (marketId, market);
         const timestamp = this.safeInteger (order, 'creation_timestamp');
         const lastUpdate = this.safeInteger (order, 'last_update_timestamp');
         const id = this.safeString (order, 'order_id');
-        let priceString = this.safeString (order, 'price');
-        if (priceString === 'market_price') {
-            // for market orders we get a literal 'market_price' string here
-            priceString = undefined;
-        }
+        const priceString = this.safeString (order, 'price');
         const averageString = this.safeString (order, 'average_price');
-        const amountString = this.safeString (order, 'amount');
-        const filledString = this.safeString (order, 'filled_amount');
+        // Invesr contracts amount is in USD which in ccxt is the cost
+        // Linear contracts amount is in corresponding cryptocurrency contracts, e.g., BTC or ETH
+        let cost = undefined;
+        // Filled amount of the order. For perpetual and futures the filled_amount is in USD units, for options and linear contracts- in units or corresponding cryptocurrency contracts, e.g., BTC or ETH.
+        let filledString = this.safeString (order, 'filled_amount');
+        let amount = this.safeString (order, 'amount');
+        if (market['inverse']) { // amount returned in USD
+            cost = Precise.stringDiv (amount, averageString);
```

Comment:
@pcriadoperez i don't think this is right and as you can see the cost is None here:
![image](https://user-images.githubusercontent.com/43336371/200116159-e283a28e-f743-497d-bf3b-323f43a75848.png)


if market['inverse'] I think we can directly assign `cost = this.safeString(order, 'amount')`

Reply:
great catch! It was diving by zero and setting the cost to None. I added a check for it. In this case the cost is zero, because the filled amount is zero.
```
ccxt deribit createOrder BTC/USD:BTC limit buy 10 25000
{
  info: {
    web: false,
    time_in_force: 'good_til_cancelled',
    risk_reducing: false,
    replaced: false,
    reduce_only: false,
    profit_loss: '0.0',
    price: '25000.0',
    post_only: false,
    order_type: 'limit',
    order_state: 'open',
    order_id: '51165205898',
    mmp: false,
    max_show: '10.0',
    last_update_timestamp: '1667749967949',
    label: '',
    is_liquidation: false,
    instrument_name: 'BTC-PERPETUAL',
    filled_amount: '0.0',
    direction: 'sell',
    creation_timestamp: '1667749967949',
    commission: '0.0',
    average_price: '0.0',
    api: true,
    amount: '10.0',
    trades: []
  },
  id: '51165205898',
  clientOrderId: undefined,
  timestamp: 1667749967949,
  datetime: '2022-11-06T15:52:47.949Z',
  lastTradeTimestamp: undefined,
  symbol: 'BTC/USD:BTC',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  side: 'sell',
  price: 25000,
  stopPrice: undefined,
  amount: undefined,
  cost: 0,
  average: undefined,
  filled: 0,
  remaining: undefined,
  status: 'open',
  fee: { cost: 0, currency: 'BTC' },
  trades: [],
  fees: [ { cost: 0, currency: 'BTC' } ]
}
```

Reply:
@pcriadoperez humm not quite right yet 
![image](https://user-images.githubusercontent.com/43336371/200281005-e0b72245-3241-4ea6-8460-b9d762568d5c.png)

Reply:
My bad I thought amount was always in base currency, but read again the docs and realized that for inverse contracts we use the quote currency. I'll fix it. Sorry for the back and forth

Reply:
@pcriadoperez no worries 🙂

Reply:
@pcriadoperez sorry to bother you again 😅  but I still get the wrong value 
![image](https://user-images.githubusercontent.com/43336371/200825034-136ffe0f-7078-45cb-93e4-55f2faa2b192.png)

can you check it out?

Reply:
I think that is correct. From what I understand.
For inverse contracts:
- amount, price and filled: quote currency (USD)
- cost: in base currency (BTC)

For linear contracts:
 - amount: base currency (BTC)
 - price and cost in quote currency (USD)
 
 From our docs:
 The amounts in an inverse contracts are quoted as if they were traded USD/BTC, however the price is still quoted terms of BTC/USD. The formula for the profit and loss of a inverse contract is (1/markPrice - 1/price) * contracts. The profit and loss and collateral will now be quoted in BTC, and the number of contracts are quoted in USD.

Reply:
@pcriadoperez yeah u're right, according to our docs "the number of contracts are quoted in USD," but I'm not sure this is the strategy applied in most exchanges rn. Nonetheless, I will merge it, and we can reevaluate it later

Reply:
@carlosmiei for me it was also very confusing. At the beginning I assumed amount was always in base, and price and cost would always be in settle currency, regardless if linear or inverse. But from the docs looks like I was wrong and this is the way.